### DISCUSSION TITLE
Can't download YOUTUBE videos

### DISCUSSION THREAD

**Discussion by tubefan** [aclap-dev/video-downloadhelper#1159] [Language & Translation]
### Your question

I can't download YOUTUBE videos, for example: **https://www.youtube.com/watch?v=5NcJvi5TYEk**, and get error messages.
I'm using Windows 10 and Firefox browser.
My VDH version:  9.1.0.42 (stable) - VdhCoApp 2.0.19
Are there new versions or how can I solve the problem?

### Addon & CoApp version and details

VDH version 9.1.0.42 (stable)
VdhCoApp 2.0.19

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.4037, Firefox 129.0.1 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20.

The Keith Richards video downloaded fine for me.  I have 2 thoughts.  First, try the VDH beta.  Second, make sure you are not logged in on YouTube.

I have a third thought.  This isn't the right area of the forum for this discussion.  It should be in regular old `Help and Q&A`, not `Language & Translation`.

**Reply by tubefan** [reply]
Hi & thank you!

VDH 9.1.0.41 beta doesn’t work.
Where can I download CoApp 2.0.20 ?

I’m not logged in on YouTube

Sorry for chosen the wrong area - can you put me in the right one or can I do that?  (How?)

**Reply by tubefan** [reply]
I have installed VdhCoApp 2.0.20, unfortunately the download still does not work.

**Reply by GrampaWildWilly** [reply]
CoApp 2.0.20 is, apparently, not yet ready for prime time.  I got it because I saw it was there but Paul warned me not to push it too hard.  It should not make any difference anyway.  There is a CoApp repository here on GitHub.  You can find it easily enough by analyzing the URLs for the CoApp you can find in the Wiki.  Or just click the aclap-dev link you can see at the top of every page of this forum.

I'm sure you have not posted your full `Copy information to clipboard` stuff because it's long & you want to save some space.  Please post all of that anyway.  Do feel free to edit it by removing your E-mail address & the part of your VDH license key that it normally includes.  Otherwise, please post all of it.

Also, when you say it doesn't work, I'm going to have to see some screenshots of what you are doing.  I asked another user in discussion #1166 to do something similar.  I posted my own screenshots of what I want.  Coincidentally, my example shows me downloading a YouTube video.  Look over there for my example & post something here along the same lines showing what you do.

As for the location of this discussion, I can't move it.  I'm not sure such a function exists even for superusers like Paul.  This is here now.  We're stuck with it.  My comment was more for future reference.  Next time you open a new discussion, just do it in the right place.

**Reply by tubefan** [reply]
**First of all: I accidentally clicked on "enable new UI". How do I get back?**

**Reply by jcfrog** [reply]
Hi,
this should help: https://github.com/aclap-dev/video-downloadhelper/wiki/V9-interface

**Reply by tubefan** [reply]
**_Should I go back to the previous VDH or CoApp versions?_**
__________________________________________________________________
version: 9.1.0.41
channel: beta
build date: 2024-08-02
build options: linuxlic: false, noyt: false, browser: firefox
lang: de
license: accepted
key (only 16 first characters): 
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:129.0) Gecko/20100101 Firefox/129.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.20",
  "binary": "
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "
  "converterBinary": "
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Reply by GrampaWildWilly** [reply]
Now you're running pretty much the same thing I am.  But that's only part of what I asked for.  You need to show us screenshots of what you do when you try to download from YouTube.  Screenshots.  Saying it doesn't work isn't enough information for anybody to diagnose the issue.  I download stuff off YouTube with that same setup all the time without incident.  So you need to show us, don't tell us, show us exactly what you're doing.  Be sure to include the URLs of the examples you are going to show us as plain text in your post.  That's just to make it easy for other folks to visit the same videos & try them for ourselves.

**Reply by tubefan** [reply]
For example, I try to download this video via "Download and convert page"

Do you need more information (e.g. error messages)?



URL:   **https://www.youtube.com/watch?v=5NcJvi5TYEk**

![grafik](https://github.com/user-attachments/assets/5dbdcf73-e23d-4d3a-b895-3cd70504cbce)

![grafik](https://github.com/user-attachments/assets/1e356cee-8918-4c5c-b784-7769ac89bd27)

**Reply by tubefan** [reply]
_I add the error message I get:_

SideDownload error:

[out#0/mp4 @ 0000019523377ec0] Error opening output C:\Keith Richards - I'm Waiting For The Man (Lou Reed Cover) (O.mp4: Permission denied

Error opening output file C:\Keith Richards - I'm Waiting For The Man (Lou Reed Cover) (O.mp4.

Error opening output files: Permission denied

**Reply by GrampaWildWilly** [reply]
Oh.  Download and Convert.  Using the legacy UI.  Oh boy.  If I recall, @paulrouget Paul has not migrated the old convert logic forward to VDH 9.  The function is still there in the legacy UI but it doesn't work.  The program logic that used to be behind that UI element simply doesn't exist any more.  Seems to me there were some discussions on that topic here in this forum.  You might search for them.  If memory serves, and it very well might not, Paul decided he was not going to migrate the convert function forward.  You are going to have to quit trying to do that in VDH.  Download with VDH but convert with some other tool.

**Reply by tubefan** [reply]
Ok, it seems to work. But why is the Download & Convert feature still in the drop-down menu? As seen in my case, it can cause problems/irritations.
Anyway, thank you very much for the great and fast help!!

Should I keep the current versions (VDH 9.1.0.41 beta, CoApp 2.0.20) or do you recommend reverting to the previously used versions (VDH version 9.1.0.42 stable, VdhCoApp 2.0.19)?

**Reply by GrampaWildWilly** [reply]
When it comes to what's in the legacy UI, I wouldn't know.  Like I said, I'm just another user.  My suspicion is Paul provided a way for people to use the old UI on the new version to provide a transition period for people to get used to the new UI.  I believe -- I emphasize I am speculating here, I have no inside information on this -- that Paul is going to deimplement the legacy UI at some point.  The new UI is much better.  I personally fell in love with it immediately the first time I saw it.  You can do very nearly everything from the keyboard now.  I found the old interface far too mouse intensive.  The operation of settings works so much better now.  The listing of variants is much cleaner.  And on & on.  People who are clinging to the legacy UI really ought to start trying to get used to the new UI.  It only stands to reason that the legacy UI is going to go away at some point.  Better to get used to the new UI sooner than later.

Since I believe the legacy UI is going to disappear at some point, I suspect Paul was less than careful about connecting all the elements in the legacy UI to new program logic.  The convert function appears to be one such case.  I would not be surprised if somebody stumbles on other similar omissions.  I would expect that Paul would just laugh at any suggestions about making things in the legacy UI work better.  I strongly recommend you & everybody start using the new UI.  If you are wrestling with something in the new UI that seems to be weird, post about it.  I'm sure somebody here can help you deal with the weirdness.

I would keep the newer versions.  The VDH beta is not some experimental  version full of bugs like many other applications might be.  The VDH beta is simply an interim release that includes some bug fixes.  Everybody should be using the beta.  The beta is eventually going to be promoted to general availability status, perhaps after a few more betas.  I should think everybody would want bug fixes early rather than wait some unknown delay for a general availability version.  That is, everybody using Firefox, of course.  It seems pretty obvious that you wouldn't want to hit some bug only to be told that it had been corrected in the beta.  Save yourself the trouble.  At least with the beta, you know that any bugs you hit really are new.  Paul would be willing to work on such reports.  If you're using anything older, he would probably start his problem diagnosis by recommending you first get the beta & try again.  That's what I would do.

As for CoApp 2.0.20, despite Paul's warning that it isn't ready for prime time, I've been using it & I haven't hit any issues.  I've even been using the ffmpeg supplied with 2.0.20 separate from VDH & it has worked fine for me so far.  I'm not encouraging anybody to upgrade to it.  But since you already have it, you might as well keep it.

**Reply by GrampaWildWilly** [reply]
What conversion are you trying to do?
